### ISSUE TITLE
Ошибка ACCESS_VIOLATION в QuikSharp.lua

### ISSUE THREAD

**Issue by sel1** [finsight/QUIKSharp#281]
После обновления до QUIK 8.10.1.1 ошибка в QuikSharp.lua:
Critical error ACCESS_VIOLATION in script C:\Program Files\QUIK_VTB\lua\QuikSharp.lua
В лог ничего не пишется
![2020-11-23 18_36_57](https://user-images.githubusercontent.com/7151891/99968435-f2773e00-2dba-11eb-978a-6eabd6346f21.png)

**Comment by Pr0phet1c** [comment]
19-значные заявки с нами уже давно (относительно). Это точно не при чем.
Попробуйте перенести файлы Lua из папки с квиком куда-нибудь отдельно, с гарантированным отсутствием проблем с правами доступа

**Comment by sel1** [comment]
Перенес LUA на другой диск, доступ точно есть, ничего не поменялось.
При запуске QUIK от имени администратора валится с ошибкой
![image](https://user-images.githubusercontent.com/7151891/100064782-c27c7900-2e54-11eb-8b18-867913bb4157.png)

Какие еще могут быть варианты?

**Comment by Pr0phet1c** [comment]
Обратиться в тех.поддержку ВТБ, с вопросом "Почему падает терминал?" Уж эта проблема точно не связана с работой библиотеки

**Comment by buybackoff** [comment]
Тут https://github.com/finsight/QUIKSharp/blob/158e05d20a939a1c25e28d1eb8afb4454bcfb46b/src/QuikSharp/lua/QuikSharp.lua#L19

8.10 парсится как 8.1 и не те бинарники используются 🤦‍♂️

**Comment by Pr0phet1c** [comment]
@buybackoff , как думаете, если полученный результат умножать на 100 и сравнивать со значениями 800 и 850, сработает?

**Comment by buybackoff** [comment]
Да, major * 100 + minor. Только нужно получить две части major/minor
отдельно как числа.

**Comment by Pr0phet1c** [comment]
> Да, major * 100 + minor. Только нужно получить две части major/minor отдельно как числа.

С этим сложнее... 
С ходу, решение в голову не приходит.

**Comment by buybackoff** [comment]
Не помню из головы, что возвращает string.split. Мы его много где используем

local major, minor = ... string split by dot ... если tuple
local version =  major * 100 + minor

или

local split =  ... string split by dot ... если table
local version =  split[1]   * 100 +  split[2] -- в Луа индексация с 1

**Comment by buybackoff** [comment]
Вижу уже есть коммит, но он неправильный. Строка получается "8.10", или 8 целых и 10 сотых, что дает число 8.1 или 810, если умножить на 100.

**Comment by Pr0phet1c** [comment]
Да, я уже переделал. Возможно, не очень изящно, но работает.

**Comment by buybackoff** [comment]
Супер! Спасибо!

**Comment by JaktensTid** [comment]
У кого такая ошибка падает попробуйте после строк

`if quikVersion == nil then
  message("QUIK# cannot detect QUIK version", 3)
  return
  else
  libPath = "\\clibs"
  end`

в QuikSharp.lua засетать quikVersion = 8.5

**Comment by vsemnado** [comment]
в версии 8.13 такая же беда. 
Как вариант можно вообще отключить проверку версии, закомментировав строки:
`	-- if quikVersion >= 8.5 then`
`        -- libPath = libPath .. "64\\53_"..linkage.."\\"`
`	-- elseif quikVersion >= 8 then`
`        -- libPath = libPath .. "64\\5.1_"..linkage.."\\"`
`	-- else`
`		-- libPath = "\\clibs\\5.1_"..linkage.."\\"`
`	-- end`
А оставить только один путь:
`libPath = libPath .. "64\\53_"..linkage.."\\"`

**Comment by Pr0phet1c** [comment]
Обновитесь. Код в этом файле уже давно другой.

**Comment by JaktensTid** [comment]
> Обновитесь. Код в этом файле уже давно другой.

Странно, я последнюю версию библиотеки себе поставил

**Comment by Pr0phet1c** [comment]
> > Обновитесь. Код в этом файле уже давно другой.
> 
> Странно, я последнюю версию библиотеки себе поставил

Я не могу знать что и как Вы ставили себе, но я точно знаю, что кусок кода, который был приведен в последнем комменте, устаревший.

Сейчас (в актуальной версии) этот кусок выглядит так:

	if quikVersion >= 805 then
        libPath = libPath .. "64\\53_"..linkage.."\\"
	elseif quikVersion >= 800 then
        libPath = libPath .. "64\\5.1_"..linkage.."\\"
	else
		libPath = "\\clibs\\5.1_"..linkage.."\\"
	end

**Comment by Camill13** [comment]
> Обновитесь. Код в этом файле уже давно другой.
Если папка "LUA" создана актуальным Nuget-пакетом, то там все еще старый код, который сбоит при версиях >= 8.10 
Если же брать ту же папку из скомпилированных примеров - то уже исправлено.
